README.md

```diff
@@ -55,6 +55,73 @@ test.all(common)
 
 A valid (read: that follows this interface) IPFS core implementation, must expose the following API.
 
+## Files
+
+### `add`
+
+> Add files and data to IPFS.
+
+##### `Go` **WIP**
+
+##### `JavaScript` - ipfs.files.add(data, [callback])
+
+Where `data` may be
+
+- an array of objects, each of the form
+```js
+{
+  path: '/tmp/myfile.txt',
+  stream: fs.createReadStream('/tmp/myfile.txt')
+}
+```
+- an array of objects, each of the form
+```js
+{
+  path: '/tmp/myfile.txt',
+  content: new Buffer('hello')
+}
+```
+- a `Buffer` instance
+- a Readable stream
+
+`callback` must follow `function (err, res) {}` signature, where `err` is an
+error if the operation was not successful. If no value `data` is provided, `res`
+will be a Duplex stream, to which tuples like the above two object formats can
+be written and [DAGNode][] objects will be outputted. Otherwise, `res` will be
+an array of [DAGNode][]s.
+
+If no `callback` is passed, a promise is returned.
```

Comment:
What does the Promise resolve to? An array of DAGNodes?

We should add examples for the returns
```javascript
ipfs.files.add(path).then((res) => console.log(res)) // --> [DAGNode1, DAGNode2, ...] ???
```